### PR TITLE
Move to generic DataFeedValueField template injection

### PR DESCRIPTION
**Pull request by Nytra**
This will generate DataFeedValueField templates using SyncMemberEditorBuilder.Build for any engine primitive type or Type type as needed

Closes https://github.com/ResoniteModdingGroup/MonkeyLoader.GamePacks.Resonite/issues/26

### CODE REVIEW

File: MonkeyLoader.Resonite.Integration/Configuration/SettingsDataFeedInjector.cs [hunk 1]
```diff
@@ -622,6 +671,11 @@ private static DataFeedValueField<T> GenerateValueField<T>(IReadOnlyList<string>
             InitBase(valueField, path, configKey);
             valueField.InitSetupValue(field => field.SyncWithConfigKey(configKey, ConfigKeyChangeLabel));
 
+            if (GenericTypesAttribute.GetTypes(GenericTypesAttribute.Group.EnginePrimitives).Contains(typeof(T)) || typeof(T) == typeof(Type))
```
Comment (Banane9) [inline MonkeyLoader.Resonite.Integration/Configuration/SettingsDataFeedInjector.cs:674]: You can replace the first part in the if with `Coder.IsEnginePrimitive(typeof(T))`

File: MonkeyLoader.Resonite.Integration/Configuration/SettingsDataFeedInjector.cs [hunk 2]
```diff
@@ -622,6 +671,11 @@ private static DataFeedValueField<T> GenerateValueField<T>(IReadOnlyList<string>
...
+            {
+                _cachedDataFeedItemMapper?.RunSynchronously(() => EnsureDataFeedValueFieldTemplate(_cachedDataFeedItemMapper, typeof(T)));
```
Comment (Banane9) [inline MonkeyLoader.Resonite.Integration/Configuration/SettingsDataFeedInjector.cs:676]: Also consider using a variable for the type of T if it's used multiple times

File: MonkeyLoader.Resonite.Integration/Configuration/SettingsDataFeedInjector.cs [hunk 3]
```diff
@@ -535,8 +585,7 @@ private static bool EnumeratePrefix(SettingsDataFeed __instance, IReadOnlyList<s
                     break;
             }
 
-            var mapper = __instance.Slot.GetComponent((DataFeedItemMapper m) => m.Mappings.Count > 1);
-            mapper?.RunSynchronously(() => EnsureColorXTemplate(mapper));
+            _cachedDataFeedItemMapper = __instance.Slot.GetComponent((DataFeedItemMapper m) => m.Mappings.Count > 1);
```
Comment (Banane9) [inline MonkeyLoader.Resonite.Integration/Configuration/SettingsDataFeedInjector.cs:588]: Should this be `>= 1`? Could also use `Mappings.Any()`.

I think I'd use an explicit generic parameter and an implicit lambda, but that's ultimately not super important.

**Review by Banane9 — CHANGES_REQUESTED** [review]

**Comment by Banane9** [comment]
Looks good otherwise 👍

File: MonkeyLoader.Resonite.Integration/Configuration/SettingsDataFeedInjector.cs (hunk 3 above)
Comment (Nytra) [reply MonkeyLoader.Resonite.Integration/Configuration/SettingsDataFeedInjector.cs:588]: There are two DataFeedItemMappers in the Settings facet, one of them has 1 mapping and needs to be ignored, so it needs to be greater than 1.

**Comment by Nytra** [comment]
Changes made

**Review by Banane9 — APPROVED** [review]

**Pull request merged**